### PR TITLE
Add context to translation payload when dealing with TextPlugin

### PR DESCRIPTION
**Pull request by filwaitman**

### CODE REVIEW

File: djangocms_text_ckeditor/cms_plugins.py [hunk 1]
```diff
@@ -197,6 +202,31 @@ def do_post_copy(self, instance, source_map):
...
+            field = django_settings.DJANGOCMS_TRANSLATIONS_CONF[obj.plugin_type]['text_field_child_label']
+            content = getattr(obj, field)
+            return plugin_to_tag(obj, content)
+
+        content = _plugin_tags_to_html(plugin_data[field], output_func=_render_plugin_with_content)
+        subplugins_within_this_content = plugin_tags_to_id_list(content)
+        return content, subplugins_within_this_content
+
+    @staticmethod
+    def get_translation_children_content(content, plugin):
```
Comment (czpython) [inline djangocms_text_ckeditor/cms_plugins.py]: `set_translation_import_content`

File: djangocms_text_ckeditor/cms_plugins.py [hunk 2]
```diff
@@ -197,6 +202,31 @@ def do_post_copy(self, instance, source_map):
         new_text = replace_plugin_tags(instance.body, ids_map)
         self.model.objects.filter(pk=instance.pk).update(body=new_text)
 
+    @staticmethod
+    def get_translation_content(field, plugin_data):
```
Comment (czpython) [inline djangocms_text_ckeditor/cms_plugins.py]: get_translation_import_content

File: djangocms_text_ckeditor/test_app/models.py [hunk 3]
```diff
@@ -1,8 +1,27 @@
...
 from djangocms_text_ckeditor.fields import HTMLField
 
 
 class SimpleText(models.Model):
     text = HTMLField(blank=True)
+
+
+@python_2_unicode_compatible
+class DummyLink(CMSPlugin):
+    cmsplugin_ptr = models.OneToOneField(
```
Comment (czpython) [inline djangocms_text_ckeditor/test_app/models.py]: not needed

File: test_requirements.txt [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+djangocms-link>=2.1.2
+
+Pillow
+html5lib>=0.999999
+
+tox>=2.9.1
+coverage>=4.4.2
+flake8>=3.0.4
+
+-e git+ssh://git@github.com/divio/djangocms-transfer.git@master#egg=djangocms-transfer
```
Comment (czpython) [inline test_requirements.txt]: hmmmm
this might be an issue on travis

File: .gitignore [hunk 5]
```diff
@@ -1,16 +1,108 @@
...
+target/
+
+# Jupyter Notebook
+.ipynb_checkpoints
+
+# pyenv
+.python-version
+
+# celery beat schedule file
+celerybeat-schedule
```
[the remark this answers is not in the document]
Comment (filwaitman) [reply .gitignore]: for sanity, I'm using the [default .gitignore for python porvided by github](https://github.com/github/gitignore/blob/master/Python.gitignore).  =]

File: test_requirements.txt (hunk 4 above)
Comment (filwaitman) [reply test_requirements.txt]: Any ideas here?

File: djangocms_text_ckeditor/cms_plugins.py [hunk 6]
```diff
@@ -197,6 +202,31 @@ def do_post_copy(self, instance, source_map):
...
+        def _render_plugin_with_content(obj, match):
+            field = django_settings.DJANGOCMS_TRANSLATIONS_CONF[obj.plugin_type]['text_field_child_label']
```
[the remark this answers is not in the document]
Comment (filwaitman) [reply djangocms_text_ckeditor/cms_plugins.py]: should I (1) duplicate the method in this project or (2) assume djangocms-translations is installed in this context?

I would go for (2).

File: djangocms_text_ckeditor/tests/test_plugin.py [hunk 7]
```diff
@@ -782,8 +796,117 @@ def test_text_plugin_xss(self):
...
+                    "<div onload='do_evil_stuff();'>divcontent</div><a href='javascript:do_evil_stuff()'>acontent</a>"
+                )
             }
             response = self.client.post(endpoint, data)
             self.assertEqual(response.status_code, 200)
             self.assertEqual(self.reload(plugin).body, '<div>divcontent</div><a>acontent</a>')
+
+
+@unittest.skipUnless(
+    HAS_DJANGOCMS_TRANSLATIONS and HAS_DJANGOCMS_TRANSFER and X,
```
[the remark this answers is not in the document]
Comment (filwaitman) [reply djangocms_text_ckeditor/tests/test_plugin.py]: X is a stupid mistake. Shame on me.  =P

**Pull request merged**